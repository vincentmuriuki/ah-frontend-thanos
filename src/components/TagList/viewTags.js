import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getTag } from '../../actions/tagsActions';
import TagList from './TagList';


export class Tags extends Component {
  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(getTag());
  }

  render() {
    const { tags } = this.props;
    return (<TagList tags={tags} key={tags.id} />);
  }
}

Tags.propTypes = {
  tags: PropTypes.arrayOf(PropTypes.object),
  dispatch: PropTypes.func.isRequired,
};
Tags.defaultProps = {
  tags: [{}],
};
export const mapStateToProps = ({ tagsReducer }) => ({ tags: tagsReducer.tags });
export default connect(mapStateToProps)(Tags);
