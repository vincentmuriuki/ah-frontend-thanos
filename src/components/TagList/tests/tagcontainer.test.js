import React from 'react';
import ConfigureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { mount } from 'enzyme';
import { Tags } from '../viewTags';

const initialState = { tagsReducer: { tags: [] } };
const mockStore = ConfigureMockStore([thunk]);


it('should render correctly', () => {
  const wrapper = mount(<Tags
    dispatch={jest.fn}
    getTag={jest.fn}
    store={mockStore(initialState)}
  />);
  expect(wrapper).toMatchSnapshot();
});
