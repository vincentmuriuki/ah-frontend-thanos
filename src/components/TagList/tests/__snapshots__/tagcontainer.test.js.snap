// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Tags
  dispatch={[Function]}
  getTag={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  tags={
    Array [
      Object {},
    ]
  }
>
  <TagList
    tags={
      Array [
        Object {},
      ]
    }
  >
    <div
      className="badge-pill h4 px-3 tag"
      key="0"
    />
  </TagList>
</Tags>
`;
